src/backend/distributed/connection/connection_management.c

```diff
@@ -616,7 +616,17 @@ ShutdownConnection(MultiConnection *connection)
 	if (PQstatus(connection->pgConn) == CONNECTION_OK &&
 		PQtransactionStatus(connection->pgConn) == PQTRANS_ACTIVE)
 	{
-		SendCancelationRequest(connection);
+		bool sentCancel = SendCancelationRequest(connection);
```

Comment:
I think the changes in `SendCancelationRequest ` makes a lot of sense, we should be doing that, especially considering he pgbouncer's behaviour.

One question that @marcocitus raised in past, do we really need  `SendCancelationRequest ` in `ShutdownConnection `? I mean, even if a remote query is running over the connection, wouldn't `PqFinish()` already cancel/terminate the connection? Might be worth understanding that and maybe getting rid of this part completely?

Reply:
My comment was mainly that we use ShutdownConnection in various failure scenarios (including various time outs) where there seems to be no value in using ShutdownConnection over CitusPQFinish given that there should be nothing to cancel. We do need to send a cancellation if a query could still be running, since PQfinish might not end the query.

Reply:
related discussion on pgbouncer: https://github.com/pgbouncer/pgbouncer/issues/63

Seems postgres doesn't stop queries when client disconnects.